UnrealPlugin/ThirdPersonMPGSDKSetup.md

```diff
@@ -0,0 +1,316 @@
+
+### Project Setup
+
+Once your project has enabled server mode, you will have a <projectname>Server.Target.cs file.
+
+Result should similar to:
+```csharp 
+public class <projectname>ServerTarget : TargetRules
+{
+    public <projectname>ServerTarget( TargetInfo Target) : base(Target)
+    {
+        Type = TargetType.Server;
+        DefaultBuildSettings = BuildSettingsVersion.V2;
+        ExtraModuleNames.AddRange( new string[] { "<projectname>" } );
+
+	// You may have additional configuration based on your server needs
+    }
+}
+```
+
+For Windows builds, you may need to add these optional configuations:
+```csharp
+DisablePlugins.Add("WMFMediaPlayer");
+DisablePlugins.Add("AsyncLoadingScreen"); //if you are using this plugin
+DisablePlugins.Add("WindowsMoviePlayer");
+DisablePlugins.Add("MediaFoundationMediaPlayer");
+```
+NOTE: These configurations are invalid for a Linux server build.
+
+### Creating/Updating the GameInstance Class
+
+#### Creating a GameInstance Class
+
+If you are in the process of creating a game from scratch and do not yet have a GameInstance class, then first follow these example instructions to create a GameInstance class. 
+
+If you are using a game that already has a GameInstance class, (for example, ShooterGame from Unreal's sample library), then move on to the section titled _**Modify the GameInstance Class**_.
+
+----
+
+In the Unreal Editor, go to Files->Create a new C++ class and select the option to "Show all classes". Then search for GameInstance. 
+By selecting it directly, everything should be generated correctly and then you can add the functions we detail below.
+
+Then close Unreal and generate project files in source build mode again.
+
+Then using Visual Studio, open those newly created files and follow instructions to modify the Game Instance class.
+
+#### Modify the Game Instance Class
+
+Locate your GameInstance class, which is most likely called something similar to [your game name]GameInstance or MyGameInstance. From now on, your game instance class will
+be denoted with [YourGameInstanceClassName].
+
+##### Modifying the GameInstance header file
+
+First, check the include statements and ensure that the following are included in the header file for your GameInstance class ([YourGameInstanceClassName].h):
+
+```cpp
+#include "CoreMinimal.h"
+#include "Engine/GameInstance.h"
+#include "MyGameInstance.generated.h"
+```
+
+Then, add the following declarations to the public section:
+(If you already have an Init() function, there is no need to include another declaration)
+
+```cpp
+public:
+
+    virtual void Init() override;
+    virtual void OnStart() override;
+```
+
+Then, add the following declarations to the protected section of methods:
+```cpp
+protected:
+
+    UFUNCTION()
+    void OnGSDKShutdown();
+    
+    UFUNCTION()
+    bool OnGSDKHealthCheck();
+};
+```
+
+##### Modifying the GameInstance Cpp file
+
+Then, locate the [YourGameInstanceClassName].cpp file. 
```

Comment:
nit: isnt't this [...] rendered as a hyperlink in markdown? [https://stackoverflow.com/questions/16027723/how-to-escape-backslash-bracket-in-markdown](https://stackoverflow.com/questions/16027723/how-to-escape-backslash-bracket-in-markdown)

Reply:
* Not all MD viewers are identical.
* You can view this file's markdown-rendered preview right here by clicking "Display the Rich Diff", which looks like a button with a page, with a bent corner
* GitHub seems to render it as shown, when there isn't a corresponding () behind it to indicate that text is a hyperlink.
* We'll need to double check it when be copy this to MSDN though, so it's a good note.
* Resolving for this PR though